internal/resources/providers/aws_cis/monitoring/filter_pattern_parser.go

```diff
@@ -0,0 +1,275 @@
+//
+//	because it can be either one or the other
+type MetricFilterPattern struct {
+	simpleExpression  //nolint:unused
+	complexExpression //nolint:unused
+}
+
+type simpleExpression struct {
+	Simple   bool
+	Left     string
+	Operator comparisonOperator
+	Right    string
+}
+
+func newSimpleExpression(left string, op comparisonOperator, right string) MetricFilterPattern {
+	return MetricFilterPattern{
+		simpleExpression: simpleExpression{
+			Simple:   true,
+			Left:     left,
+			Operator: op,
+			Right:    right,
+		},
+	}
+}
+
+type complexExpression struct {
+	Complex     bool
+	Operator    logicalOperator
+	Expressions []MetricFilterPattern
+}
+
+func newComplexExpression(op logicalOperator, exps ...MetricFilterPattern) MetricFilterPattern {
+	return MetricFilterPattern{
+		complexExpression: complexExpression{
+			Complex:     true,
+			Operator:    op,
+			Expressions: exps,
+		},
+	}
+}
+
+// parseFilterPattern receives a string with CloudWatch FilterPattern syntax and parses to an expressions
+// Known limitations:
+//   - It parses only 5 levels deep expressions. You can increase with maxDepth const. But we don't have that use case
+//   - It doesn't properly handle reserved characters (=, !=, NOT EXISTS, &&, ||) inside strings. We don't have that use case yet
+func parseFilterPattern(s string) (MetricFilterPattern, error) {
+	// remove trailing spaces and { }
+	cleanS := strings.TrimSpace(strings.TrimRight(strings.TrimLeft(strings.TrimSpace(s), "{"), "}"))
+
+	if strings.Count(s, "(") != strings.Count(s, ")") {
+		return MetricFilterPattern{}, errors.New("broken parenthesis")
+	}
+
+	return safeParse(cleanS, 0)
+}
+
+func safeParse(s string, depth int) (MetricFilterPattern, error) {
+	if depth > maxDepth {
+		return MetricFilterPattern{}, errors.New("max depth reached, can't parse this expression")
+	}
+
+	var logicalOp logicalOperator
+	expressions := make([]MetricFilterPattern, 0, 10)
+
+	buf := strings.Builder{}
+	buf.Grow(len(s))
+
+	pointer := 0
+	for len(s) > pointer {
+		r := rune(s[pointer])
+		i := pointer
+		pointer++
+
+		if r == '(' { // If it's a parenthesis opening, resolve the parenthesis
+			exp, closingParenthesisPos, err := resolveParenthesis(s, depth, i)
+			if err != nil {
+				return MetricFilterPattern{}, err
+			}
+			expressions = append(expressions, exp)
+			pointer = closingParenthesisPos + i + 1 // move pointer to the end of what has been already processed
+			continue
+		}
+
+		if _, err := buf.WriteRune(r); err != nil {
+			return MetricFilterPattern{}, fmt.Errorf("could not write rune %v", err)
+		}
+
+		tmpString := buf.String()
+		if contains, op := hasSuffixLogicalOp(tmpString); contains { // && or || marks the end of a Simple MetricFilterPattern
+			if logicalOp == "" { // set the Operator of the MetricFilterPattern without overriding it
+				logicalOp = op
+			}
+
+			if logicalOp != op {
+				return MetricFilterPattern{}, errors.New("not supported comparison with alternating logical operators")
```

Comment:
Do I understand correctly, that the parser only supports a chain of same boolean operators, so both `x AND y AND z` and `x OR y OR z` are supported, but not any combination like `x AND y OR z`?